push_docker.sh

```diff
@@ -1,42 +1,42 @@
+#! /bin/bash 
+
 DOCKER_IMAGE_FILE=$1
 REGISTRY_SERVER=$2
-REGISTRY_PORT=$3
-REGISTRY_USERNAME=$4
-REGISTRY_PASSWD=$5
-DOCKER_IMAGE_TAG=$6
+REGISTRY_USERNAME=$3
+REGISTRY_PASSWD=$4
+PLATFORM=$5
+SONIC_VERSION=$6
+DOCKER_IMAGE_TAG=$7
+
+push_it() {
+    docker tag $1 $2
+    echo "Pushing $2"
+    image_sha=$(docker push $2 | sed -n "s/.*: digest: sha256:\([0-9a-f]*\).*/\\1/p")
+    echo "Remove $2"
+    docker rmi $2 || true
+    echo "Image sha256: $image_sha"
+}
 
 set -e
+echo "Docker load $DOCKER_IMAGE_FILE"
 docker load < $DOCKER_IMAGE_FILE
 
-## Fetch the Jenkins build number if inside it
-[ ${BUILD_NUMBER} ] || {
-    echo "No BUILD_NUMBER found, setting to 0."
-    BUILD_NUMBER="0"
-}
+## Login the docker image registry server
+## Note: user name and password are passed from command line
+echo "Docker login $REGISTRY_USERNAME@$REGISTRY_SERVER"
+docker login -u $REGISTRY_USERNAME -p "$REGISTRY_PASSWD" $REGISTRY_SERVER
 
-## Prepare tag
+## Get image name
 docker_image_name=$(basename $DOCKER_IMAGE_FILE | cut -d. -f1)
-remote_image_name=$REGISTRY_SERVER:$REGISTRY_PORT/$docker_image_name:$DOCKER_IMAGE_TAG
-timestamp="$(date -u +%Y%m%d)"
-build_version="${timestamp}.bld-${BUILD_NUMBER}"
-build_remote_image_name=$REGISTRY_SERVER:$REGISTRY_PORT/$docker_image_name:$build_version
 
-## Add registry information as tag, so will push as latest
-## Add additional tag with build information
-docker tag $docker_image_name $remote_image_name
-docker tag $docker_image_name $build_remote_image_name
+remote_image_name=$REGISTRY_SERVER/sonic-dockers/$PLATFORM/$docker_image_name
```

Comment:
>sonic-dockers/$PLATFORM/ [](start = 35, length = 24)

This is also breaking change. What is the intention?

Reply:
We have same docker name for broadcom & Mellanox. So we need platform in the path to distinguish.

Reply:
For example? we already have names like `docker-syncd-mlnx`

---
In reply to: [402525553](https://github.com/Azure/sonic-buildimage/pull/4337#discussion_r402525553) [](ancestors = 402525553)

Reply:
For example? we already have names like docker-syncd-mlnx

---
In reply to: [402512143](https://github.com/Azure/sonic-buildimage/pull/4337#discussion_r402512143) [](ancestors = 402512143)

Reply:
e.g. docker_database.gz 
The name is same for both Broadcom & Mellanox.

Reply:
I see the intention. To be general, you can treat `sonic-dockers/broadcom/` as an optional argument to this script and support this feature. But do not break old behavior.

---
In reply to: [402699047](https://github.com/Azure/sonic-buildimage/pull/4337#discussion_r402699047) [](ancestors = 402699047,402512143)

Reply:
Done.